src/js/utils/error.ts

```diff
@@ -0,0 +1,33 @@
+export interface ExtendedError extends Error {
+  framesToPop?: number | undefined;
+}
+
+// Sentry Stack Parser is skipping lines not frames
+// https://github.com/getsentry/sentry-javascript/blob/739d904342aaf9327312f409952f14ceff4ae1ab/packages/utils/src/stacktrace.ts#L23
+// 1 for first line with the Error message
+const SENTRY_STACK_PARSER_OFFSET = 1;
+const REMOVE_ERROR_CREATION_FRAMES = 2 + SENTRY_STACK_PARSER_OFFSET;
+
+/**
+ * Creates synthetic trace. By default pops 2 frames - `createSyntheticError` and the caller
+ */
+export function createSyntheticError(framesToPop: number = 0): ExtendedError {
+  const error: ExtendedError = new Error();
+  error.framesToPop = framesToPop + REMOVE_ERROR_CREATION_FRAMES; // Skip createSyntheticError's own stack frame.
+  return error;
+}
+
+/**
+ * Returns the number of frames to pop from the stack trace.
+ * @param error ExtendedError
+ */
+export function getFramesToPop(error: ExtendedError): number {
+  return error.framesToPop !== undefined ? error.framesToPop : 0;
+}
+
+/**
+ * Check if `wat` is an object with string stack property.
+ */
+export function isErrorLike(wat: unknown): wat is { stack: string } {
+  return wat !== null && typeof wat === 'object' && 'stack' in wat && typeof wat.stack === 'string';
+}
```

Comment:
suggestion
```suggestion
 * Check if potentialError is an object with string stack property.
 */
Suggestion
```suggestion
export function isErrorLike(potentialError : unknown): potentialError is { stack: string } {
  return potentialError !== null && typeof potentialError === 'object' && 'stack' in potentialError && typeof potentialError .stack === 'string';
}
```
```